### DISCUSSION TITLE
Renaming PWA for Google Chrome

### DISCUSSION THREAD

**Discussion by imeidopf** [coder/code-server#6383] [Q&A] [answered]
I've recently looked into making my Samsung Galaxy Tab S6 a lightweight dev machine. I've been installing some web apps as apps, since Google Chrome offers this for PWAs. This app has the necessary bits to install the site as an app but it only gives me "code-server" as the name. How can I make it say "Code Server" or "VS Code"?

**Comment by code-asher** [comment, marked as answer]
If it helps, the correct name of this project is indeed 
code-server and not Code Server or VS Code, but if you want to 
change it, I believe you can use `--app-name` to change the name 
in the PWA manifest.

**Reply by imeidopf** [reply]
I know this is the name. I want it to display a friendly name, without the dash. I have this running in a docker container, installed through Portainer. Do I have to do something before installation? The manifest.json file I'm seeing has the code - oss.

**Reply by code-asher** [reply]
The point I was hoping to make is that Code Server is not a different/friendly version of the same name but rather simply the wrong name (like Catherine Zeta-Jones without the dash, Jan van Dijk with a capital V,  WordPress with a lowercase p, VS Code without a space in the middle, macOS with a capital M, iPad with a capital I, etc).

Code Server can also be more easily confused with VS Code Server or OpenVSCode Server, separate repositories and separate products, so brand awareness and discoverability is part of the game here.

As for the VS Code name, depending on one's goals (for example starting a SaaS) there could be legal trouble as it is [trademarked](https://code.visualstudio.com/brand).

But, if that does not sway you, no hard feelings.  I can see how the whole thing might feel arbitrary.

You can set `--app-name` either via the Docker invocation (I am not familiar with Portainer but if running directly it would look something like `docker run codercom/code-sever --app-name 'Code Server'`) or in the `~/.config/code-server/config.yaml` file (`app-name: Code Server` should do the trick).

**Comment by danopia** [comment]
I landed on this FAQ because I have too many code-servers and want to distinguish them in the Chromebook launcher bar.

Apparently, the third-party Docker image `docker.io/linuxserver/code-server` in particular doesn't seem to enable passing extra arguments to the `code-server` process. So adding e.g. `--app-name=penguin` failed to start up for me.

Instead, I edited `~/.config/code-server/config.yaml` from within the code-server to add e.g. `app-name: penguin`, and then restarted the whole process. After updating every container, I removed and re-added the PWAs. Now they are named uniquely in the launcher and I can fearlessly alt-tab between different servers. The launcher used to be extremely ambiguous before, now it's clear:

![image](https://github.com/user-attachments/assets/de385ad8-9fb5-430a-a76e-7d0997702da8)
